docs/HowToGuides/GettingStarted.md

```diff
@@ -0,0 +1,613 @@
+
+## Cloning the project
+
+1. Create a directory for the whole project:
+   ```sh
+   mkdir -p swift-project/swift
+   cd swift-project/swift
+   ```
+2. Clone the sources:
+   - Via SSH (recommended):
+     If you plan on contributing regularly, cloning over SSH provides a better
+     experience. After you've [uploaded your SSH keys to GitHub][]:
+     ```sh
+     git clone git@github.com:apple/swift.git .
+     utils/update-checkout --clone-with-ssh
+     ```
+   - Via HTTPS:
+     If you want to check out the sources as read-only,
+     or are not familiar with setting up SSH,
+     you can use HTTPS instead:
+     ```sh
+     git clone https://github.com/apple/swift.git .
+     utils/update-checkout --clone
+     ```
+   **Note:** If you've already forked the project on GitHub at this stage,
+   **do not clone your fork** to start off. We describe
+   [how to setup your fork](#setting-up-your-fork) in a subsection below.
+   <!-- Recommending against cloning the fork due to SR-13476 and SR-13505. -->
+3. Double-check that `swift`'s sibling directories are present.
+   ```sh
+   ls ..
+   ```
+   This should list directories like `llvm-project`, `swiftpm` and so on.
+4. Checkout the right branch/tag:
+   If you are building the toolchain for local development, you can skip this
+   step, as Step 2 will checkout `swift`'s `master` branch and matching
+   branches for other projects. <!-- [TODO: master-to-main-rename] -->
+   If you are building the toolchain as a one-off, it is more likely that you
+   want a specific branch or a tag, often corresponding to a specific release
+   or a specific snapshot. You can update the branch/tag for all repositories
+   as follows:
+   ```sh
+   utils/update-checkout --branch mybranchname
+   # OR
+   utils/update-checkout --tag mytagname
+   ```
+   Detailed branching information, including names for release branches, can
+   be found in [Branches.md](/docs/Branches.md).
+
+**Note:**
+The commands used in the rest of this guide assumes that the absolute path
+to your working directory is something like `/path/to/swift-project/swift`.
+Double-check that running `pwd` prints a path ending with `swift`.
+
+[uploaded your SSH keys to GitHub]: https://help.github.com/articles/adding-a-new-ssh-key-to-your-github-account/
+
+### Setting up your fork
+
+If you are building the toolchain for development and submitting patches,
+you will need to setup a GitHub fork.
+
+First fork the `apple/swift` [repository](https://github.com/apple/swift.git),
+using the "Fork" button in the web UI, near the top-right. This will create a
+repository `username/swift` for your GitHub username. Next, add it as a remote:
+```sh
+# Using 'my-remote' as a placeholder name.
+
+# If you set up SSH in step 2
+git remote add my-remote git@github.com:username/swift.git
+
+# If you used HTTPS in step 2
+git remote add my-remote https://github.com/username/swift.git
+```
+Finally, create a new branch.
```

Comment:
I think this should move to the editing stage. You don't need to create a branch to set up the fork, but you do when you're about to edit and make changes.

Reply:
I moved the whole subsection to be under the 'Editing Code' section, because you don't need to setup a fork if you're just building a release branch as a one-off.